### ISSUE TITLE
Twin-fit method interface with surrogate-gradient default

### ISSUE THREAD

**Issue by johnjosephhorton** [expectedparrot/zwill#2]
## Summary

Define `twin-fit` as an extensible digital-twin construction framework, with the surrogate-gradient method from the soft-drink methodological note as the first transparent default—not as the permanent definition of twin fitting.

The default method should:

1. accumulate observations pairing bounded respondent trait vectors with LLM choice probabilities;
2. convert K probabilities to K−1 log-odds against a declared reference option;
3. fit a ridge-regularized multivariate linear model from standardized traits to those logits;
4. reconstruct probabilities with softmax;
5. differentiate training-choice cross-entropy through the surrogate;
6. propose a clipped, trust-region-bounded trait update;
7. rerun the authoritative LLM at the proposed state;
8. compare predicted and realized loss changes; and
9. add successful and unsuccessful verification points to the next fit.

This is a **construction/fitting method**. Its diagnostics must not be presented as a validation report. A frozen fitted AgentList can subsequently enter Zwill's existing held-out `twin-validate` workflow.

## Why an interface is needed

The ridge surrogate is attractive as a default because it is inexpensive, differentiable, inspectable, and produces coefficient diagnostics. It will not be best for every LLM response surface. The artifact and command contracts should allow later methods such as:

- nonlinear differentiable surrogates;
- finite-difference or coordinate search;
- Bayesian optimization;
- trust-region methods;
- LLM-proposed diffs followed by conventional ranking; and
- ensembles that use disagreement to choose new perturbations.

Upstream schema/state/observation artifacts and downstream diff/verification artifacts should remain stable when the optimizer changes.

## Proposed method contract

Each fitting method should consume:

- approved trait schema and bounds;
- versioned respondent states;
- normalized observation rows with fixed option order;
- observed training choices;
- explicit split and round identifiers; and
- declared hyperparameters and random seeds.

It should emit:

- a versioned method artifact with training-row lineage;
- local fit and extrapolation diagnostics;
- one or more bounded proposed diffs;
- predicted loss before and after each diff; and
- enough metadata to reproduce the proposal.

Verification remains method-independent: Zwill exports candidate AgentLists, the user executes EDSL jobs externally, and Zwill imports realized LLM probabilities.

## Suggested CLI

```bash
zwill twin-fit method list
zwill twin-fit surrogate fit \
  --method ridge-multivariate-logit \
  --survey soda \
  --schema soft_drink_v1 \
  --alpha 1.0 \
  --path surrogate-round-0.json

zwill twin-fit diff propose \
  --method-artifact surrogate-round-0.json \
  --objective nll \
  --learning-rate 0.25 \
  --max-step 0.25 \
  --path diffs-round-1.jsonl
```

The current `traits` command may remain temporarily as a deprecated alias while the workflow moves to the practitioner-oriented `twin-fit` name.

## Diagnostics for the default method

- coefficient matrix with reference option clearly labeled;
- training and leave-one-trajectory/checkpoint-out probability MAE;
- rank/design coverage and zero-variance warnings;
- predicted-versus-realized changes in Brier and NLL;
- correlation and sign agreement between predicted and realized improvements;
- clipping and trust-region activation rates;
- distance from the observed perturbation support; and
- accumulated observation counts by round and split.

## Construction versus validation terminology

Use:

- construction audit;
- surrogate diagnostics;
- optimization trace;
- update verification; and
- fitted or frozen AgentList.

Reserve `validation report`, `validated twin`, and predictive-lift claims for the downstream held-out workflow.

## Acceptance criteria

- [ ] Ridge multivariate-logit is registered as the default fitting method through a method interface.
- [ ] Method artifacts record schema, option order, transformations, coefficients, scaling, hyperparameters, row lineage, and version.
- [ ] Diff generation depends only on the declared method interface.
- [ ] Verification reports are method-independent.
- [ ] Later fitting methods can be added without changing schema, state, observation, diff, or verification contracts.
- [ ] The CLI and docs distinguish fitting evidence from held-out validation evidence.
- [ ] The soda tutorial explains every mathematical transformation with a numeric example.
- [ ] A frozen fitted AgentList can be handed to the existing validation workflow without being modified.

## Relationship to #1

This issue narrows the algorithm boundary within the broader parameterized-twin implementation in #1. Issue #1 owns the end-to-end artifacts and workflow; this issue owns the fitting-method abstraction and the default surrogate-gradient implementation.
